assignments/this.js

```diff
@@ -1,26 +1,52 @@
 /* The for principles of "this";
 * in your own words. explain the four principle for the "this" keyword below.
 *
-* 1. 
-* 2. 
-* 3. 
-* 4. 
+* 1. implicit binding - look to the left of the " . " when you call/invoke the function
+* 2. Explicit binding - using "call" "apply" or "bind" to tell the "this" keyword it's context
+* 3. new binding - "new" keyword is bound to new object being constructed
+* 4. window binding - if none of the other types apply "this" defaults to the viewport (window object)
 *
 * write out a code example of each explanation above
 */
 
 // Principle 1
 
 // code example for Window Binding
+console.log(this.name)
 
 // Principle 2
 
+
 // code example for Implicit Binding
+    const MyObject = function (){
```

Comment:
This is not an example of Implicit Binding. Implicit binding takes place when `this` is used inside of an object. In this example, this is being used inside of a function. Oh and `MyObject` isn't actually an object when assigned on line 21, it is a constructor function.